### PR TITLE
Admin Generator Future: add mode to form to generate code for editing or adding

### PR DESCRIPTION
**Pull request by nsams**
When forms are only used for editing this greatly simplifies the generated code

Also add a ProductPriceForm to Demo  that is only used for editing

### CODE REVIEW

File: packages/admin/cms-admin/src/generator/future/generator.ts [hunk 1]
```diff
@@ -28,6 +28,7 @@ export type FormFieldConfig<T> = (
 export type FormConfig<T extends { __typename?: string }> = {
     type: "form";
     gqlType: T["__typename"];
+    mode?: "edit" | "add" | "editAdd";
```
Comment (johnnyomair) [inline packages/admin/cms-admin/src/generator/future/generator.ts:31]: I don't like editAdd 😬 How about `both` or `default` instead?

**Comment by johnnyomair** [comment]
Is there ever a use case for `mode="add"`? If yes, should we add one in Demo?

File: packages/admin/cms-admin/src/generator/future/generator.ts (hunk 1 above)
Comment (johnnyomair) [reply packages/admin/cms-admin/src/generator/future/generator.ts:31]: You could also remove it altogether and use `mode?: "add-only" | "edit-only"`

Comment (nsams) [reply packages/admin/cms-admin/src/generator/future/generator.ts:31]: I also don't like editAdd.

what about `"all"`?

**Comment by nsams** [comment]
> Is there ever a use case for mode="add"?

sure - when a custom form is needed for adding.

> If yes, should we add one in Demo?

hm, I'm not sure, that doesn't fit into it's use case. But it could be done.

File: packages/admin/cms-admin/src/generator/future/generator.ts (hunk 1 above)
Comment (johnnyomair) [reply packages/admin/cms-admin/src/generator/future/generator.ts:31]: > what about "all"?

Sounds good

**Pull request merged**